src/lib/mod.rs

```diff
@@ -77,59 +79,73 @@ impl HacksawResult {
     }
 }
 
-pub fn set_shape(conn: &xcb::Connection, window: xcb::Window, rects: &[xcb::Rectangle]) {
+pub fn set_shape<C: Connection>(conn: &C, window: xproto::Window, rects: &[xproto::Rectangle]) {
     shape::rectangles(
-        &conn,
-        shape::SO_SET as u8,
-        shape::SK_BOUNDING as u8,
-        0,
+        conn,
+        shape::SO::Set,
+        shape::SK::Bounding,
+        xproto::ClipOrdering::Unsorted,
         window,
         0,
         0,
         &rects,
-    );
+    )
+    .unwrap()
+    .check()
+    .unwrap();
```

Comment:
This is somewhat unsatisfying

Reply:
Yes, but its better than ignoring the result. We can do unwrap cleanup in another PR

Reply:
Alright